src/kakarot/instructions/system_operations.cairo

```diff
@@ -212,11 +212,14 @@ namespace SystemOperations {
         range_check_ptr,
         bitwise_ptr: BitwiseBuiltin*,
     }(ctx: model.ExecutionContext*) -> model.ExecutionContext* {
+        alloc_locals;
         let (ctx, call_args) = CallHelper.prepare_args(ctx=ctx, with_value=1);
-
-        // TODO: use gas_limit when init_at_address is updated
+        let remaining_gas = ctx.gas_limit - ctx.gas_used;
+        let (max_allowed_gas, _) = Helpers.div_rem(remaining_gas, 64);
+        let gas_limit = Helpers.min(call_args.gas, max_allowed_gas);
```

Comment:
can't we include this in `prepare_args` as it's the same op for all the CALLs codes?